runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/config/MqttKafkaRouteConfig.java

```diff
@@ -60,15 +60,23 @@ boolean authorized(
         return authorized.test(authorization);
     }
 
-    boolean matches(
+    public boolean matchesSubscribe(
         String topic)
     {
-        return when.isEmpty() || topic != null && when.stream().anyMatch(m -> m.matches(topic));
+        return when.isEmpty() || when.stream().anyMatch(m -> m.matchesSubscribe(topic));
+    }
+
+    public boolean matchesClient(
+        String client)
+    {
+        return !when.isEmpty() && when.stream()
+            .filter(m -> m.kind == MqttKafkaConditionMatcher.MqttKafkaConditionKind.SUBSCRIBE)
+            .allMatch(m -> m.matchesSubscribe(client));
     }
 
-    public boolean matchesTopicFilter(
+    public boolean matchesPublish(
         String topic)
     {
-        return when.isEmpty() || when.stream().anyMatch(m -> m.matches(topic));
+        return when.isEmpty() || when.stream().anyMatch(m -> m.matchesPublish(topic));
```

Comment:
This should filter by kind `PUBLISH`?

Now that we have `kind` do we need separate `matchesPublish` and `matchesSubscribe` methods? Perhaps just `matches` method and filter by kind here instead?